### ISSUE TITLE
Problem with mesh transformation

### ISSUE THREAD

**Issue by mcgratta** [firemodels/fds#7561]
This case
```
&HEAD CHID='simple_test' /

&MESH IJK=36,24,24, XB=0.0,3.6,0.0,2.4,0.0,1.2 /
&MESH IJK=36,24,24, XB=0.0,3.6,0.0,2.4,1.2,2.4 /

&TIME T_END=0.1 /

&TRNX IDERIV=0, CC=1.8, PC=1.8, MESH_NUMBER=1 /
&TRNX IDERIV=1, CC=1.8, PC=0.5, MESH_NUMBER=1 /
&TRNX IDERIV=0, CC=1.8, PC=1.8, MESH_NUMBER=2 /
&TRNX IDERIV=1, CC=1.8, PC=0.5, MESH_NUMBER=2 /

&ZONE XYZ=1,1,1 /

&TAIL /
```
worked up until PR #7237 

I can't figure out exactly what has gone wrong.

**Comment by rmcdermo** [comment]
This should be fixed with #7521.  Please test and close if it works for you.

Note that previously reported Issue #7234 seems to work, although the way I fixed this one was to basically undo the previous changes.

**Comment by mcgratta** [comment]
Works, thanks.
